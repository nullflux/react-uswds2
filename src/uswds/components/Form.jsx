import React from 'react';
import PropTypes from 'prop-types';

/**
 * The Form is used to create an HTML form for user input.
 *
 */
export default class Form extends React.Component {
  
  static propTypes = {
    action: PropTypes.string,
    children: PropTypes.node.isRequired,
    classes: PropTypes.string,
    id: PropTypes.string
  };

  static defaultProps = {
    action: '',
    classes: '',
    id: ''
  };

  render() {
    return (
      <form id={this.props.id} name={this.props.id} className={`usa-form ${this.props.classes}`} action={this.props.action}>
        {this.props.children}
      </form>
    );
  }
}
