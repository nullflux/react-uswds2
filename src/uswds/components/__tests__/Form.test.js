import React from 'react';
import { shallow } from 'enzyme';
import Form from '../Form.jsx';
import Fieldset from '../Fieldset.jsx';
import TextInput from '../TextInput.jsx';

jest.unmock('../Form.jsx');
jest.unmock('../Fieldset.jsx');
jest.unmock('../TextInput.jsx');

describe('Form', () => {
  it('should render correctly', () => {
    const component = shallow(
    <Form>
      <Fieldset legend="Name">
      <TextInput id="last-name" label="Last name" />
      </Fieldset>
    </Form>
    );
    expect(component).toMatchSnapshot();
  });
});