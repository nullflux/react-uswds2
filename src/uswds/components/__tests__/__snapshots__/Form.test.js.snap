// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  action=""
  className="usa-form "
  id=""
  name=""
>
  <Fieldset
    legend="Name"
    legendHidden={false}
  >
    <TextInput
      classes=""
      enableSpellCheck={false}
      id="last-name"
      label="Last name"
      required={false}
      type="text"
    />
  </Fieldset>
</form>
`;
